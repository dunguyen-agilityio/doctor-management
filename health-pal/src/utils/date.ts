import dayjs from 'dayjs'

export const formatTime = (time: string, separator = '.') => {
  const splitTime = time.split(':')

  return `${splitTime.slice(0, 2).join(separator)} ${parseInt(splitTime[0]) > 12 ? 'PM' : 'AM'}`
}

export const getDefaultDate = () => {
  let now = dayjs()

  while ([0, 6].includes(now.day())) {
    if (![0, 6].includes(now.day())) break
    now = now.set('date', now.get('date') + 1)
  }

  return now
}
