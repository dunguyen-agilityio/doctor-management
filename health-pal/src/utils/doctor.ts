import { DoctorData, TDoctorCard } from '@app/models/doctor'

export const formatDoctor = ({
  clinic,
  specialty,
  users_permissions_user,
  ...rest
}: DoctorData): TDoctorCard => {
  const { avatar, ...user } = users_permissions_user

  return {
    ...user,
    ...rest,
    address: clinic.address,
    avatar: avatar?.url ?? '',
    specialty: specialty.name,
  }
}
