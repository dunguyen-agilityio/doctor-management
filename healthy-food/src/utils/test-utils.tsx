import {
  RenderOptions,
  render,
  renderHook,
} from '@testing-library/react-native';

import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const AllTheProviders = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>{children}</NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (renderCallback, options) =>
  renderHook(renderCallback, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render, customRenderHook as renderHook };
