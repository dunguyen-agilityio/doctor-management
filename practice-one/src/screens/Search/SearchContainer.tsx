import React, { useCallback, useRef } from 'react';
import { TextInput, View } from 'react-native';

import { useFocusEffect } from '@react-navigation/native';

import { Categories, Loading, SearchInput } from '@/components';
import ErrorFallback from '@/components/ErrorFallback';

import { CATEGORIES } from '@/constants';

import { useFoods, useSearchQuery } from '@/hooks';
import { useFilters } from '@/hooks';

const SearchContainer = ({
  children,
  fallback = <Loading />,
}: React.PropsWithChildren<{ fallback?: React.ReactNode }>) => {
  const { query, setQuery } = useSearchQuery();
  const { filters: categories, setFilters } = useFilters();
  const { error, isLoading } = useFoods({ categories, query });
  const searchInputRef = useRef<TextInput>(null);

  useFocusEffect(
    useCallback(() => {
      return () => {
        setQuery('');
        setFilters([]);
        if (searchInputRef.current) {
          searchInputRef.current.clear();
        }
      };
    }, [setFilters, setQuery]),
  );

  const handlePressTag = useCallback(
    (id: string) => {
      setFilters((prev) => {
        const newTags = prev.includes(id)
          ? prev.filter((item) => item !== id)
          : [...prev, id];

        return newTags;
      });
    },
    [setFilters],
  );

  if (error) return <ErrorFallback error={error} />;

  return (
    <View style={{ flex: 1 }}>
      <SearchInput ref={searchInputRef} onChangeText={setQuery} />
      <Categories
        categories={CATEGORIES}
        onSelect={handlePressTag}
        values={categories}
      />
      {children}
      {isLoading && fallback}
    </View>
  );
};

export default SearchContainer;
