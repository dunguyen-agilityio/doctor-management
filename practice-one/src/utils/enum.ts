export const isInEnum = <T extends Record<string, string | number>>(
  obj: T,
  key: unknown = null,
): key is keyof T => {
  if (key == null) return false;
  return Object.keys(obj).includes(String(key));
};

export const isArrayEqual = <T>(arr1: T[], arr2: T[]) =>
  arr1.length == arr2.length && arr1.every((item) => arr2.includes(item));
