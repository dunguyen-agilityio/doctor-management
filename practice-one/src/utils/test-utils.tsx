import { RenderOptions, render } from '@testing-library/react-native';

import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const AllTheProviders = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient({
    defaultOptions: { queries: { retry: false } },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>{children}</NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
