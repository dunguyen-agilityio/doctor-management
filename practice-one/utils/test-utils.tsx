import { ReactElement } from 'react';

import { NavigationContainer } from '@react-navigation/native';

import { RenderOptions, render } from '@testing-library/react-native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
