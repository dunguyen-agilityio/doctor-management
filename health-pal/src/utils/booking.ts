import { BookingData } from '@/models/booking'

import { createDayjs } from './date'

export const formatBooking = ({ doctor, date, documentId, id, time, type }: BookingData) => {
  const { users_permissions_user, clinic, documentId: doctorDocId, id: doctorId } = doctor

  return {
    date: createDayjs(date),
    documentId,
    id,
    time,
    type,
    doctorName: users_permissions_user.name,
    doctorAvatar: users_permissions_user.avatar?.url,
    address: clinic.address,
    specialty: doctor.specialty.name,
    doctorId,
    doctorDocId,
  }
}
