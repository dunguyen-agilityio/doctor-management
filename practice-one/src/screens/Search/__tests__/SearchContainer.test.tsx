import SearchContainer from '@/screens/Search/SearchContainer';

import { render } from '@/utils/test-utils';

import FiltersProvider from '@/contexts/filters';
import SearchProvider from '@/contexts/search';

describe('SearchContainer', () => {
  it('renders FoodContainer match snapshot', () => {
    const tree = render(
      <SearchProvider>
        <FiltersProvider>
          <SearchContainer />
        </FiltersProvider>
      </SearchProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
