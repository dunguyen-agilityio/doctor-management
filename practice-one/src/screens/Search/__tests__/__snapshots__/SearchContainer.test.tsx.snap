// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer renders FoodsContainer match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
  testID="foods-container"
>
  <View
    style={
      {
        "flex": 1,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "position": "absolute",
            "width": "100%",
            "zIndex": 10,
          },
          {
            "marginTop": undefined,
          },
        ]
      }
      testID="loading-indicator"
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/images/logo.png",
            }
          }
          style={
            {
              "height": 122,
              "width": 122,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
